### PR TITLE
[master] Porting #49148 to master

### PR DESCRIPTION
**Pull request by garethgreenaway**
Porting #49148 to master

### CODE REVIEW

File: salt/grains/core.py [hunk 1]
```diff
@@ -144,14 +144,16 @@ def _linux_cpudata():
...
+            grains['num_cpus'] = 0
             for line in _fp:
                 comps = line.split(':')
                 if not len(comps) > 1:
                     continue
                 key = comps[0].strip()
                 val = comps[1].strip()
                 if key == 'processor':
-                    grains['num_cpus'] = int(val) + 1
+                    log.debug('=== processor %s %s ===', key, val)
```
[the remark this answers is not in the document]
Comment (dwoz) [reply salt/grains/core.py]: I removed the log message while fixing a merge conflict

**Review by s0undt3ch — DISMISSED** [review]

**Review by Ch3LL — CHANGES_REQUESTED** [review]
this port is missing this line: https://github.com/saltstack/salt/pull/49148/files#diff-1d38435cf2b529355dec413e02b3fdc26f1af1d89f119891c68c4e9cf5630062R143

is this on purpose?

also this will require a changelog

**Comment by garethgreenaway** [comment]
@Ch3LL Good catch.  Looks like something that just got missed in the origin PR.

**Comment by Ch3LL** [comment]
thanks, looks like this will also need a changelog

**Comment by garethgreenaway** [comment]
@Ch3LL done.

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**